### DISCUSSION TITLE
Environment Variables

### DISCUSSION THREAD

**Discussion by alkorlos** [11ty/eleventy#3123] [Q&A]
Node.js from v20.6.0 have [built-in .env file support](https://nodejs.org/en/blog/release/v20.6.0#built-in-env-file-support), can call `node --env-file=config.env`, very comfortably.

If the way is to use this together with eleventy? If now call `eleventy --env-file=.env.development` there will be an error
```
[11ty] Eleventy CLI Fatal Error: (more in DEBUG output)
[11ty] We don’t know what '--env-file=config.env' is. Use --help to see the list of supported commands. (via EleventyCommandCheckError)
```

**Comment by Snapstromegon** [comment]
The way you passed them in, you pass them as a parameter to the eleventy module, but you'd want it as a node parameter. Sadly, as I see it right now, neither npm exec nor npx have support for this parameter.
I would've expected `npx --node-options="--env-file=.env" eleventy` to work, but it doesn't.

**Reply by alkorlos** [reply]
Thanks, that is to `npm exec` or `npx`. As far as I understand, you are not sure whether this is possible or not, if someone writes at once, can close it.
Cheers.
